omnisend/src/connection/components/connection-page-layout.js

```diff
@@ -0,0 +1,68 @@
+import ConnectionLogos from './connection-logos';
+import ConnectionFeatures from './connection-features';
+import ConnectionSteps from './connection-steps';
+import { useState } from '@wordpress/element';
+import { Notice, Flex, Spinner } from '@wordpress/components';
+
+const ConnectionPageLayout = () => {
+	const [error, setError] = useState(null);
+	const [loading, setLoading] = useState(null);
+
+	const connect = (apiKey) => {
+		const fd = new FormData();
+		fd.append('api_key', apiKey);
+		setLoading(true);
+		fetch('/wp-json/omnisend/v1/connect', {
+			method: 'POST',
+			body: fd,
+		})
+			.then((response) => response.json())
+			.then((data) => {
+				if (data.success) {
+					location.reload();
+				}
+				if (data.error) {
+					setError(data.error);
+					setLoading(false);
+				}
+			})
+			.catch((e) => {
+				setError(e.message || e);
```

Comment:
The error sholuld be clear for the user, wont this result in some unclear timeout error?